components/tools/OmeroWeb/omeroweb/webclient/views.py

```diff
@@ -189,108 +189,72 @@ def login(request):
-                            del request.session['active_group']
-                    if request.session.get('user_id'):
-                        # always want to revert to logged-in user
-                        del request.session['user_id']
-                    if request.session.get('server_settings'):
-                        # always clean when logging in
-                        del request.session['server_settings']
-                    # do we ned to display server version ?
-                    # server_version = conn.getServerVersion()
-                    if request.POST.get('noredirect'):
-                        return HttpResponse('OK')
-                    url = request.GET.get("url")
-                    if url is None or len(url) == 0:
-                        try:
-                            url = parse_url(settings.LOGIN_REDIRECT)
-                        except:
-                            url = reverse("webindex")
-                    return HttpResponseRedirect(url)
-                else:
-                    error = "This user is not active."
 
-        if not connector.is_server_up(useragent):
-            error = "Server is not responding, please contact administrator."
-        elif not settings.CHECK_VERSION:
-            error = ("Connection not available, please check your"
-                     " credentials and version compatibility.")
-        else:
-            if not compatible:
-                error = ("Client version does not match server,"
-                         " please contact administrator.")
+    def get(self, request, *args, **kwargs):
+        return self._handleNotLoggedIn(request, *args, **kwargs)
+
+    def _handleLoggedIn(self, request, conn, connector, *args, **kwargs):
+
+        # webclient has various state that needs cleaning up...
+        # if 'active_group' remains in session from previous
+        # login, check it's valid for this user
+        if request.session.get('active_group'):
+            if (request.session.get('active_group') not in
+                    conn.getEventContext().memberOfGroups):
+                del request.session['active_group']
+        if request.session.get('user_id'):
+            # always want to revert to logged-in user
+            del request.session['user_id']
+        if request.session.get('server_settings'):
+            # always clean when logging in
+            del request.session['server_settings']
+        # do we ned to display server version ?
+        # server_version = conn.getServerVersion()
+        if request.POST.get('noredirect'):
+            return HttpResponse('OK')
+        url = request.GET.get("url")
+        if url is None or len(url) == 0:
+            try:
+                url = parse_url(settings.LOGIN_REDIRECT)
+            except:
+                url = reverse("webindex")
+        return HttpResponseRedirect(url)
+
+    def _handleNotLoggedIn(self, request, error=None, form=None, **kwargs):
```

Comment:
Done. ```error``` and ```form``` are now optional keyword arguments on the superclass ```on_not_logged_in()```  https://github.com/openmicroscopy/openmicroscopy/pull/4708/commits/33203889f95e80773ca3dbae736b144bcd96cd90